syft/pkg/cataloger/golang/scaffolding_parser.go

```diff
@@ -0,0 +1,216 @@
+package golang
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/common/command"
+	"github.com/anchore/syft/syft/source"
+)
+
+const (
+	GoSumFileName = "go.sum"
+	GoCommand     = "go"
+)
+
+func parseScaffolding(location source.Location, reader io.Reader) ([]*pkg.Package, []artifact.Relationship, error) {
+	currentFilepath := string(location.VisibleRealPath())
+	parser := scaffoldingParser{
+		currentFilepath: currentFilepath,
+		currentFileDir:  filepath.Dir(currentFilepath),
+		reader:          reader,
+		command:         GoCommand,
+	}
+	return parser.parse()
+}
+
+type scaffoldingParser struct {
+	currentFilepath string
+	currentFileDir  string
+	reader          io.Reader
+	command         string
+}
+
+func (p *scaffoldingParser) parse() ([]*pkg.Package, []artifact.Relationship, error) {
+	// lock file isn't existed
+	if _, err := os.Stat(filepath.Join(filepath.Dir(p.currentFilepath), GoSumFileName)); err != nil {
+		return parseGoMod(p.currentFilepath, p.reader)
+	}
+
+	log.Infof("parsing golang dependency file %s by scaffolding", p.currentFilepath)
+	var ctx, cancel = context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
Why do we need the timeout?

Reply:
Just prevent command runs for a long time. Even if we have not encountered the situation of being stuck now.